// Copyright 2025 The Ray Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//  http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto3";

import "google/protobuf/timestamp.proto";

package ray.rpc.events;

// Message containing the definition of a node, as observed via GCS.
// The message is expected to be emitted once per node creation.
//
// For runtime information associated with this event, see NodeLifecycleEvent.
message NodeDefinitionEvent {
  bytes node_id = 1;
  string node_ip_address = 2;
  map<string, string> labels = 3;
  google.protobuf.Timestamp start_timestamp = 4;
}
